### PR TITLE
Add license checker

### PR DESCRIPTION
**Pull request by leonplust**
Add GitHub Action that checks license line in each file for issue #22 . Because I don't know which file or directory should be excluded. I have set it to check all the .cpp and .h files in all directories. I have tested run it and below is part of the result of the test run.
Screen Shot 2021-04-26 at 1 12 08 AM

### CODE REVIEW

File: .licenserc.json [hunk 1]
```diff
@@ -0,0 +1,14 @@
+{
+    "**/*.{h, cpp}": [
```
Comment (BradSwain) [inline .licenserc.json:2]: Can you ignore the `examples` and `tests/data` directories?

**Review by BradSwain — CHANGES_REQUESTED** [review]

**Comment by BradSwain** [comment]
Looks great!
I think the `examples` and `tests/data` directories are the only ones we want to ignore.
Other than that looks good to me.

I will let @funemy also take a look to make sure I did not miss anything.

**Comment by leonplust** [comment]
I have made the changes to ignore tests/data and examples/. Now all the files are passing the license check.

**Comment by leonplust** [comment]
I have made the changes to ignore tests/data and examples/. Now all the files are passing the license check.

**Comment by funemy** [comment]
The change looks good to me in general.

Just to make sure, do we also want license for files like Dockerfile and those pthread test cases (those created by us)?
If not I think this PR is ready to go @BradSwain

**Comment by BradSwain** [comment]
@funemy I think only the core race detection code and test case logic needs to include the license header, but I will check and we and we can update it later.

@leonplust Thank you for your contribution! I will merge the change.

**Review by BradSwain — APPROVED** [review]

**Comment by funemy** [comment]
@BradSwain ok, in that case I think the checker is added correctly.

@leonplust Thanks for the contribution, this is a neat PR.

**Comment by leonplust** [comment]
You are welcome! I'm glad that I can help!

**Pull request merged**